langlib/lang.regexp/src/main/java/org/ballerinalang/langlib/regexp/Find.java

```diff
@@ -38,6 +39,10 @@
 public class Find {
 
     public static BArray find(BRegexpValue regExp, BString str, long startIndex) {
+        if (isEmptyRegexp(regExp)) {
```

Comment:
We can remove this logic in this PR

Reply:
Will be removing this with #39795 once the Runtime case is handled as well.